src/Thoth.Json/Decode.fs

```diff
@@ -543,12 +543,21 @@ let private mixedArray msg (decoders: Decoder<obj>[]) (values: obj[]): Result<ob
             | Error _ -> acc
             | Ok result -> decoder value |> Result.map (fun v -> v::result))
 
-let rec private autoDecodeRecordsAndUnions (t: System.Type): Decoder<obj> =
+let inline private lowerCase (s: string) =
+    s |> Seq.mapi (fun i c -> match i with | 0 -> (System.Char.ToLower c) | _ -> c)  |> System.String.Concat
```

Comment:
This is traversing through all the chars just to lower the first one, slicing fits here nicely: `(string s.[0]).ToLower() + s.[1..]`